app/src/main/java/hu/vmiklos/plees_tracker/Preferences.kt

```diff
@@ -20,6 +30,76 @@ class Preferences : PreferenceFragmentCompat() {
             it.summary = path
         }
     }
+
+    override fun onPreferenceTreeClick(preference: Preference): Boolean {
+        val sharedPreferences = context?.let { PreferenceManager.getDefaultSharedPreferences(it) }
+
+        if (preference.key == "enable_dnd") { // If someone toggled the enable_dnd preference
+            if (sharedPreferences != null) {
+                if (sharedPreferences.getBoolean("enable_dnd", false)) {
+                    val activity = activity ?: return super.onPreferenceTreeClick(preference)
+                    val noteServ = Context.NOTIFICATION_SERVICE // 100 char limit
+                    val dndManager = activity.getSystemService(noteServ) as NotificationManager
+                    val hasPermission = dndManager.isNotificationPolicyAccessGranted
+
+                    if (!hasPermission) {
+                        // If we don't have permissions for DND
+                        AlertDialog.Builder(context)
+                            .setTitle(R.string.settings_enable_dnd_q_title)
+                            .setMessage(R.string.settings_enable_dnd_q_message)
+                            .setPositiveButton(R.string.settings_enable_dnd_q_ok) { _, _ ->
+                                val setting_id = Settings.ACTION_NOTIFICATION_POLICY_ACCESS_SETTINGS
+                                val intent = Intent(setting_id)
+                                startActivity(intent)
+                            }
+                            .setNegativeButton(R.string.settings_enable_dnd_q_cancel) { _, _ ->
+                                val editor = sharedPreferences.edit()
+                                if (editor != null) {
+                                    editor.putBoolean("enable_dnd", false)
+                                    editor.apply()
+                                } else {
+                                    Log.wtf(TAG, "editor is null")
```

Comment:
Existing code just use error in such a case, you could do the same here for consistency.

Reply:
I used `Log.wtf` mainly because of the fact that editor shouldn't be null at all. As according to Android documentation:

> `Log.wtf`:
> What a Terrible Failure: Report a condition that should never happen

But I will change it to `Log.e` for consistency